src/go/docker-credential-none/helpers.go

```diff
@@ -0,0 +1,81 @@
+package dcnone
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+	"syscall"
+
+	"github.com/docker/docker-credential-helpers/credentials"
+	dockerconfig "github.com/docker/docker/cli/config"
+)
+
+type dockerConfigType map[string]interface{}
+
+func getParsedConfig() (dockerConfigType, error) {
+	dockerConfig := make(dockerConfigType)
+	contents, err := os.ReadFile(configFile)
+	if err != nil {
+		if errors.Is(err, syscall.ENOENT) {
+			// Time to create a new config (or return no data)
+			return dockerConfig, nil
+		}
+		return dockerConfig, err
+	}
+	err = json.Unmarshal(contents, &dockerConfig)
+	if err != nil {
+		return dockerConfig, fmt.Errorf("reading config file %s: %s", configFile, err)
+	}
+	return dockerConfig, nil
+}
+
+func saveParsedConfig(config *dockerConfigType) error {
+	contents, err := json.MarshalIndent(config, "", "  ")
+	if err != nil {
+		return err
+	}
+	scratchFile, err := os.CreateTemp(dockerconfig.Dir(), "tmpconfig.json")
```

Comment:
I'm curious as to why you create a temporary file and move it to the final location, rather than just writing it to the final location. Can `os.WriteFile` do a partial write and then return an error or something?

Reply:
It's standard practice (at least for us) for updating a file. If `os.WriteFile` fails for some reason, we haven't lost the original file. And we can only hope that if `os.Rename` fails for some reason, it doesn't trash the original file.